clusterloader2/pkg/state/resource_state.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state
+
+import (
+	"fmt"
+	"strconv"
+	"sync"
+)
+
+// ResourceIdentifier is a unique identifier for resource
+type ResourceIdentifier struct {
```

Comment:
nit: ResourceIdentifier may be a misleading name.
Maybe simply ResourceType ?

Reply:
TBH I prefer `ResourceIdentifier`, cause it an identifier. It just happened to be the same thing as resource type at this point.
I can see a case when we decide to add (for example) namespace to this struct.  It would be still identifier, but not a resource type.

Reply:
But Resource is something different - it's an instance of an object for example.

I'm fine with leaving "Identifier" suffix, but it's not a resource.
Since I don't think we will add anything to it anytime soon, I would go with ResourceTypeIdentifier, and if at some point we will have to add something, we will change the name.

Reply:
I get your point. Renamed to `ResourceTypeIdentifier`.